Add October 2022 release notes (#760)
returntocorp/semgrep-docs: docs/release-notes.md

Welcome to Semgrep release notes. This document gives an overview of the changes, additions, and selected fixes. Release notes published since April 2022 include Semgrep CLI, CI, and Semgrep App updates. Release notes published since April 2022 encompass all versions released during the month.

## October 2022

### Semgrep Supply Chain

> r2c now offers a new product: Semgrep Supply Chain, a software composition analysis solution. Semgrep Supply Chain is a high-signal dependency scanner that detects reachable vulnerabilities in open source, third-party libraries in your code. Reach out to sales@r2c.dev to learn more.

Reviewer: ```suggestion
r2c now offers a new product: Semgrep Supply Chain. Semgrep Supply Chain is a high-signal dependency scanner that detects reachable vulnerabilities in open source, third-party libraries in your code. Learn more about [Semgrep Supply Chain](https://semgrep.dev/products/semgrep-supply-chain).
```

### Semgrep App

#### Additions

- New demo project allows you to try out Semgrep App workflows. If your organization in Semgrep App does not have any projects assigned in the [Projects](https://semgrep.dev/orgs/-/projects) page, you can add the new demo project by clicking **Explore a demo project**.
- You can now triage through PR comments, for more information see [Ignoring findings through comments](https://semgrep.dev/docs/semgrep-app/findings/#ignoring-findings-through-comments) documentation.
- Semgrep Playground now displays its version number. To see the exact version that Semgrep Playground uses, click the <i class="fa-regular fa-ellipsis-vertical"></i> three-dot button, and then see the version number after the <i class="fa-solid fa-code-commit"></i> icon.

    ![Semgrep Playground version number.](/img/semgrep-app-latest-version.png "Semgrep Playground version number.")

#### Changes

> - Removing a rule automatically removes all findings that came from the removed rule. If you add a removed rule back, then you need to re-scan your code to get the findings from the previously removed rule again. For more information, see [Triaging findings](https://semgrep.dev/docs/semgrep-app/findings/#triaging-findings).

Reviewer: @raghavjain3 Should we be noisier in the community Slack about this?

Reviewer 2: @pabloest @adamkvitek we can message in the community slack! 

This is an extension of the existing behavior - right now you have to rescan to see the findings removed. With the change, they are automatically removed without rescan.

Author: Previously, when you removed a rule you had to rescan the code to remove findings produced by the rule. With this change, findings produced by the removed rule are now automatically removed without rescanning.

        pattern-sources:
          - pattern: tainted(...)
        severity: WARNING
        ```

    - The `taint_assume_safe_functions`, makes Semgrep assume that function calls do **not** propagate taint from their arguments to their output. Otherwise, Semgrep always assumes that functions may propagate taint. This is intended to replace **not-conflicting** sanitizers (added in v0.69.0, for more information, see [Minimizing false positives via sanitizers](https://semgrep.dev/docs/writing-rules/data-flow/taint-mode/#minimizing-false-positives-via-sanitizers)) in the future. This option is still experimental and needs to be complemented by other changes in future releases. To enable this option, include the `taint_assume_safe_functions: true` under the `options` key. For example:
        ```yaml    
        rules:
          - id: tainted
            languages:
              - php
                message: Match
            mode: taint
            options:
              taint_assume_safe_functions: true
            pattern-sinks:
              - pattern: sink(...)
            pattern-sources:
              - pattern: tainted(...)
            severity: WARNING

>         ```

Author: @pabloest we mentioned both of these under https://semgrep.dev/docs/writing-rules/rule-syntax/#options 
But sometimes documenting how to do something in release notes can also be beneficial. If you do not find it helpful let me know and I will remove the code snippets in the release notes and link to rule-syntax section linked above.

![Screenshot 2022-10-31 at 17 13 20](https://user-images.githubusercontent.com/26669307/199055949-e6c855a8-ee28-4288-88db-32e480d6dad1.png)

Reviewer: I think if they will be not changing over time (I expect they won’t), then this looks good to me.

Author: They may change, they are experimental...

Author: But I may remove them from release notes once they will be updated.

    Previously, if Semgrep found a definition such as `wrapper() { return taint_source; }`, it recognized that `wrapper` was propagating taint from `taint_source`. If the code included something as `sink(wrapper())`, Semgrep flagged it. However, for Semgrep to match such code, a `wrapper` had to be defined earlier in the source file before its use. DeepSemgrep is able to handle this case and many more.
    If you relied on this feature, it can generally be worked around using for example:
   ```yaml
   pattern-sources:
    - patterns:
        - pattern-inside: |
            $FUNC(...) {
              ...
              return tainted_source;
            }
            ...
        - pattern: $FUNC(...)
   ```

- Disabled Bloom filter optimization by default, due to undesired interactions with constant and symbolic propagation, while it appears to not provide a net major performance benefit. If you do notice a significant drop in performance after this change, please let us know.

### Semgrep in CI

#### Changes 

> - Previously, Semgrep in CI environments overrode user-defined environment variables with values it detected from the CI provider. Now, user-defined environment variables take precedence (override) Semgrep's detected values. By enabling you to override CI variables, you are able to troubleshoot issues such as hyperlinks to code in the Findings page and receiving comments in pull or merge requests.

Reviewer: ```suggestion
- Previously, Semgrep overrode user-defined environment variables with values it detected from the CI provider. Now, user-defined environment variables take precedence (override) Semgrep's detected values. By enabling you to override CI variables, you are able to troubleshoot issues such as hyperlinks to code in the Findings page and receiving comments in pull or merge requests.
```
